src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlIds.java

```diff
@@ -219,63 +205,33 @@ HtmlId forMember(TypeElement typeElement, VariableElement member) {
     }
 
     /**
-     * Returns an id for the erasure of an executable element,
-     * or {@code null} if there are no type variables in the signature.
-     *
-     * For backward compatibility, include an anchor using the erasures of the
-     * parameters.  NOTE:  We won't need this method anymore after we fix
-     * {@code @see} tags so that they use the type instead of the erasure.
+     * Returns a string id for the erasure of an executable element.
      *
-     * @param executableElement the element to anchor to
-     * @return the 1.4.x style anchor for the executable element
+     * @param executable the element to anchor to
      */
-    protected HtmlId forErasure(ExecutableElement executableElement) {
-        final StringBuilder buf = new StringBuilder(executableElement.getSimpleName().toString());
-        buf.append("(");
-        List<? extends VariableElement> parameters = executableElement.getParameters();
-        boolean foundTypeVariable = false;
-        for (int i = 0; i < parameters.size(); i++) {
-            if (i > 0) {
-                buf.append(",");
+    private String forErasure(TypeElement site, ExecutableElement executable) {
+        var parameterTypes = ((ExecutableType) utils.typeUtils
+                .erasure(utils.asInstantiatedMethodType(site, executable)))
+                .getParameterTypes();
+        var stv = new SimpleTypeVisitor9<String, Void>() {
+            @Override
+            public String visitArray(ArrayType t, Void p) {
+                return visit(t.getComponentType()) + utils.getDimension(t);
```

Comment:
Obviously, `utils.getDimension()` harkens back to the old (JDK 8-era) doclet, but even the current implementation of that method could use some TLC. But that's not your problem here.